std/list.mgn

```diff
@@ -25,6 +27,22 @@ rangeWithNext := \start.next.end. rangeWithStep start (minus next start) end
  */
 rangeWithStep := Y (\rec.start.step.end. (le start end) (cons start (rec (plus start step) step end)) nil)
 
+/* Maps function `f` over the elements of list `xs`
+ */
+map := \xs.f. Y (\mapRec.xs.f. (isnil xs) [] (cons (f (head xs)) (mapRec (tail xs) f))) xs f
+
+/* Creates a list containing `n` times `x`
+ */
+replicate := \x.n. Y (\replicateRec.n.x.i. (lt i n) (cons x (replicateRec n x (succ i))) []) n x 0
+
+foldl := \xs.zero.op. Y (\foldlRec.xs.acc.op. (isnil xs) acc (foldlRec (tail xs) (op acc (head xs)) op)) xs zero op
```

Comment:
Also `op` is already is the scope of inner function. No need to thread it through the entire iteration:

`\op.zero.xs. Y (\foldlRec.acc.xs. ???...`